### PR TITLE
fix(warehouse): increase async job timeout

### PR DESCRIPTION
**Pull request by achettyiitr**
# Description

- Increase warehouse async job timeout to 5 minutes. Since 10 seconds interval is definitely sufficient to run the Query.
- Use a shorthand name for `AsyncJobWhT` as a instead of asyncWhJob.
- Moved the global logger inside the `AsyncJobWhT` itself.
- using WithConfig to populate the configs.

## Notion Ticket

https://www.notion.so/rudderstacks/Warehouse-Async-Job-Framework-Sync-79a575f429e540d79e9094651b30d7e8

## Security

- [x] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

**Review by chaitanyapraveen — APPROVED** [review]
LGTM

**Review by Jayachand — APPROVED** [review]

File: warehouse/jobs/types.go [hunk 1]
```diff
@@ -91,5 +91,5 @@ const (
 	MaxQueryRetries       int = 3
 	RetryTimeInterval         = 10 * time.Second
 	MaxAttemptsPerJob     int = 3
-	WhAsyncJobTimeOut         = 10 * time.Second
```
[the remark this answers is not in the document]
Comment (achettyiitr) [reply warehouse/jobs/types.go:94]: Even @Jayachand mentioned the same thing.

[the remark this answers is not in the document]
Comment (achettyiitr) [reply warehouse/jobs/types.go:94]: Added these.

File: warehouse/jobs/handlers.go [hunk 2]
```diff
@@ -18,36 +18,36 @@ import (
 )
 
 // AddWarehouseJobHandler The following handler gets called for adding async
-func (asyncWhJob *AsyncJobWhT) AddWarehouseJobHandler(w http.ResponseWriter, r *http.Request) {
-	pkgLogger.Info("[WH-Jobs] Got Async Job Add Request")
-	pkgLogger.LogRequest(r)
+func (a *AsyncJobWhT) AddWarehouseJobHandler(w http.ResponseWriter, r *http.Request) {
```
Comment (Jayachand) [inline warehouse/jobs/handlers.go:21]: `a` looks like too generic. rename to `asyncJob` ??

Comment (achettyiitr) [reply warehouse/jobs/handlers.go:21]: [The name of a method's receiver should be a reflection of its identity; often a one or two letter abbreviation of its type suffices (such as "c" or "cl" for "Client"). ](https://github.com/golang/go/wiki/CodeReviewComments#receiver-names)

**Review by lvrach — APPROVED** [review]

**Review by chaitanyapraveen — APPROVED** [review]
LGTM!

**Pull request merged**